### DISCUSSION TITLE
Generate Rotor from existing CAD-File (.step)

### DISCUSSION THREAD

**Discussion by ilm71** [byuflowlab/FLOWUnsteady#79] [Q&A]
Dear All, 

I wanted to ask how it is possible to generate a rotor in FlowUnsteady from an existing CAD-airfoil shape including the spinner and the two blade geometries. Is there a way to import .step files into FlowUnsteady directly or is it necessary to find a workaround to extract the sweepdistr, chorddistr. etc. for different radius ratios separately and import them as a .csv file in the "generate rotor" function?

Many thanks for your answer!

**Comment by ilm71** [comment]
does anyone have a solution for including geometries like from .step-files?

**Reply by Turbulence1116** [reply]
I wonder the same

**Reply by jayjiejie** [reply]
I got the same question. If I would like to perform a propeller open water test simulation with a propeller CAD file (stl, step, etc.) with integrated propeller hub and blades. Is there a way to load the entire geometry into FLOWUnsteady?

A relevant question is if FLOWUnsteady takes any types of mesh files?

Thank you very much.

**Reply by jayjiejie** [reply]
@Turbulence1116 Saw your post on the propeller wing interaction (https://github.com/byuflowlab/FLOWUnsteady/discussions/98), I am wondering if you managed to re-create your propeller geometry in FLOWUnsteady. If so, would you be able to share a bit of experience how to do it? Thank you very much.

**Comment by EdoAlvarezR** [comment]
Hi All! 

The public releases of FLOWUnsteady only assume flat surfaces (so, no thick bodies) as briefly discussed in the docs: [LINK](https://flow.byu.edu/FLOWUnsteady/examples/blownwing-asm/). However, in the [flowpanel_prototype branch](https://github.com/byuflowlab/FLOWUnsteady/tree/flowpanel-prototype) we have integrated a 3D panel method for the simulation of thick bodies, which is discussed in the AVIATION 2023 paper: [LINK](https://scholarsarchive.byu.edu/cgi/viewcontent.cgi?article=7676&context=facpub). However, the panel method so far assume a structured grid and [generated through GeometricTools
](https://flow.byu.edu/FLOWPanel.jl/stable/geometry/gridgeneration/).

@cibinjoseph has been working on integrating OpenVSP into FLOWUnsteady and hopefully in the next few months I'll rework FLOWPanel to be able to take unstructured grids generated through OpenVSP!

**Reply by jayjiejie** [reply]
Hi Ed Alvarez

Thank you very much for your reply. Just wonder if there are any instructions about generating the mesh using the FLOWPanel? Is the generated mesh equivalent to any other common mesh types that can be generated using a meshing software?

It will also be appreciated to have some tutorials on how to use this structured grid with FLOWUnsteady branch - flowpanel_protoytpe.

I did see the propeller non-symmetric loft example, which uses blade sectional profiles to generate the 3D panel using the non-symmetric loft. I am wondering if the same can be done to a marine propeller using in a similar manner with the sectional profile. However, these profiles need to be joined along a curve instead of a straight line to form a skewed blade, which may require all x_pos, y_pos and z_pos to define the location of the chord with the appropriate twist?

Is it also possible to include a propeller hub cone with the 3D blade panels if the above step is feasible with the FLOWPanel?

Thank you very much.
Kind regards,
Jie

**Reply by EdoAlvarezR** [reply]
>  Just wonder if there are any instructions about generating the mesh using the FLOWPanel? 

I'd recommend checking the docs regarding geometry in FLOWPanel: [LINK](https://flow.byu.edu/FLOWPanel.jl/stable/geometry/gridgeneration)

> It will also be appreciated to have some tutorials on how to use this structured grid with FLOWUnsteady branch - flowpanel_protoytpe.

For the moment, the panel method this is a feature under development, so no support is provided until we are ready to merge it into main. Feel free to venture and use it in your own though---we would appreciate you helping in the development!

**Reply by jayjiejie** [reply]
Hi Ed Alvarez

I tried to follow the propeller example to build a propeller from its sectional profiles, chord distributions, pitch distribution, sweep distribution, height distribution, etc. I am looking at the apc10x7*.csv files in the datafile/rotor directory. I am a bit confused as there are different number of r/R data points among those files. For example, chord distribution are given at different r/R locations compared to the height distribution. Will there be any issue in this case? I am wondering if there is any documentation about how to build the propeller from such information? What is the minimum information that we need to re-create the propeller geometry in FLOWUnsteady?

Thank you very much.

Jie

**Reply by cibinjoseph** [reply]
@jayjiejie I believe FLOWUnsteady uses an interpolation when you give different r/R values in those propeller files. it shouldn't be a problem. You can always compare both the step geometry with the geometry generated from FLOWUnsteady by running a few computation steps and looking at the rotor geometry in the solutions.
